api/users/tests/utils/test_mailer_lite.py

```diff
@@ -0,0 +1,160 @@
+    )
+    user3 = FFAdminUser.objects.create(
+        email="test_user3", first_name="test", last_name="test"
+    )
+
+    users = [user1, user2, user3]
+    mocker.patch("users.utils.mailer_lite.MAX_BATCH_SIZE", 2)
+
+    mocked_request = mocker.patch("users.utils.mailer_lite.requests")
+
+    # When
+    with BatchSubscribe() as batch:
+        for user in users:
+            batch.subscribe(user)
+    # Then
+    # assert that batch_send is called(using requests made) twice, first time for
+    # hitting the maximum limit and second time
+    # for exiting the context manager
+    assert mocked_request.post.call_count == 2
+
+
+@pytest.mark.django_db
+def test_batch_subscribe__subscribe_populates_batch_correctly(mocker):
+    # Given
+    user1 = FFAdminUser.objects.create(
+        email="test_user1", first_name="test", last_name="test"
+    )
+    user2 = FFAdminUser.objects.create(
+        email="test_user2", first_name="test", last_name="test"
+    )
+    users = [user1, user2]
+    # When
+    with BatchSubscribe() as batch:
+        for user in users:
+            batch.subscribe(user)
+        # Then
+        len(batch._batch) == len(users)
+        assert batch._batch[0]["body"]["email"] == users[0].email
+        assert batch._batch[1]["body"]["email"] == users[1].email
+
+
+@pytest.mark.django_db
+def test_get_request_body_from_user_with_paid_organisations(mocker):
+    # Given
+    user = FFAdminUser.objects.create(
+        email="test_user1", first_name="test", last_name="test"
+    )
+
+    organisation = Organisation.objects.create(name="Test org")
+    Subscription.objects.create(organisation=organisation)
+    user.add_organisation(organisation)
+
+    # When
+    data = _get_request_body_from_user(user)
+
+    # Then
+    assert data == {
+        "email": user.email,
+        "name": "test test",
+        "fields": {"is_paid": True},
+    }
+
+
+def test_batch_subscribe_batch_send_makes_correct_post_request(mocker, settings):
+
+    # Given
+    mocked_request = mocker.patch("users.utils.mailer_lite.requests")
+    mocked_headers = mocker.patch(
+        "users.utils.mailer_lite.MailerLiteBaseClient.request_headers",
+    )
+
+    base_url = "http//localhost/mailer/test/"
+    settings.MAILERLITE_BASE_URL = base_url
+    resource = "batch"
+
+    batch = BatchSubscribe()
+    test_batch_data = [1, 2, 3]
+
+    mocker.patch.object(batch, "_batch", test_batch_data.copy())
+
+    # When
+    batch.batch_send()
+    # Then
+    mocked_request.post.assert_called_with(
+        base_url + resource,
+        data=json.dumps({"requests": test_batch_data}),
+        headers=mocked_headers,
+    )
+    assert batch._batch == []
+
+
+def test_mailer_lite_base_client_request_headers(settings):
+    # Given
+    api_key = "test_key"
+    # First, let's update settings
+    settings.MAILERLITE_API_KEY = api_key
+    # Reload the module to use updated settings
+    from users.utils import mailer_lite
```

Comment:
@matthewelwell  not sure if there is any better way to do this?

Reply:
... do we need this test at all? Since it's just a constant, I'd say probably not.

Reply:
In case we made a typo there? But then where do we stop.

Reply:
Haha exactly. I think it's fine to not test this.